client/src/components/Markdown/Elements/HistoryDatasetDisplay.vue

```diff
@@ -54,53 +54,66 @@
-                                    v-slot="{ result: metaData, loading: metaLoading, error: metaError }"
-                                    :url="metaUrl">
-                                    <LoadingSpan v-if="metaLoading" message="Loading Metadata" />
-                                    <div v-else-if="metaError">{{ metaError }}</div>
+            <UrlDataProvider v-slot="{ result: itemContent, loading: datasetLoading, error }" :url="itemUrl">
+                <LoadingSpan v-if="datasetLoading" message="Loading Dataset" />
+                <LoadingSpan v-else-if="loading" message="Loading Datatypes" />
+                <div v-else-if="error">{{ error }}</div>
+                <div v-else-if="!datatypesMapper">Datatypes not loaded.</div>
+                <div v-else :class="contentClass">
+                    <b-embed
+                        v-if="datatypesMapper.isSubTypeOfAny(datasetType, ['pdf', 'html'])"
+                        type="iframe"
+                        aspect="16by9"
+                        :src="displayUrl" />
+                    <HistoryDatasetAsImage
+                        v-else-if="datatypesMapper.isSubTypeOfAny(datasetType, ['galaxy.datatypes.images.Image'])"
+                        :args="args" />
+                    <div v-else-if="itemContent.item_data">
+                        <div v-if="datatypesMapper.isSubTypeOfAny(datasetType, ['tabular'])">
+                            <UrlDataProvider
+                                v-slot="{ result: metaData, loading: metaLoading, error: metaError }"
+                                :url="metaUrl">
+                                <LoadingSpan v-if="metaLoading" message="Loading Metadata" />
+                                <div v-else-if="metaError">{{ metaError }}</div>
+                                <div v-else>
                                     <b-table
-                                        v-else
+                                        id="tabular-dataset-table"
+                                        sticky-header
+                                        thead-tr-class="sticky-top"
                                         striped
                                         hover
-                                        :fields="getFields(metaData)"
-                                        :items="getItems(itemContent.item_data, metaData)" />
-                                </UrlDataProvider>
-                            </div>
-                            <pre v-else>
-                                <code class="word-wrap-normal">{{ itemContent.item_data }}</code>
-                            </pre>
+                                        :per-page="perPage"
+                                        :current-page="currentPage"
+                                        :fields="Object.freeze(getFields(metaData))"
+                                        :items="Object.freeze(getItems(itemContent.item_data, metaData))" />
```

Comment:
```suggestion
                                        :fields="getFields(metaData)"
                                        :items="getItems(itemContent.item_data, metaData)" />
```
I would not optimize things that don't need optimization